src/service/viewport-impl.js

```diff
@@ -1093,22 +1137,37 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
+  hideViewerHeader(transient, lastPaddingTop) {
+    if (!transient) {
+      this.updatePaddingTop(0);
+    } else {
+      // Add extra paddingTop to make the content stay at the same position
+      // when the hiding header operation is transient
+      onDocumentReady(this.win.document, doc => {
+        const existingPaddingTop =
+            this.win./*OK*/getComputedStyle(doc.body)['padding-top'] || '0';
+        doc.body.style.paddingTop =
```

Comment:
When we hide the header, we remove the `borderTop`. We want to make the content stays at the same place after removing `borderTop`. I tried `scrollIntoView` function provided by the browser to 'adjust the scroll position' but it doesn't work while the viewport is being scrolled. So we add extra `paddingTop` after we remove `borderTop`.